### DISCUSSION TITLE
How to specify particular Exchange for particular event?

### DISCUSSION THREAD

**Discussion by remotenode** [BrighterCommand/Brighter#1571] [Q&A]
Hi @iancooper,

My microservice needs posting events to different exchanges, is it possible with Brighter? I noticed that in a startup class it's possible to specify only default exchange. I use RabbitMQ with InMemoryOutbox

Thanks in advance.

**Comment by remotenode** [comment]
Hi @preardon, maybe you could help?

**Comment by iancooper** [comment]
@artem-avanesov It's probable that I don't understand the question, but you configure the exchange as part of the RmqMessagingGatewayConnection.

If you look at this example, it is lines 50-54: https://github.com/BrighterCommand/Brighter/blob/master/samples/RMQTaskQueueSamples/GreetingsSender/Program.cs

**Reply by iancooper** [reply]
I guess it is possible that you mean, can I post a message to multiple exchanges simultaneously?

We don't support that at the moment. We assume that your intent is to use an exchange for routing messages for a given routing key to subscribers on one exchange.

**Reply by iancooper** [reply]
In theory, you can solve this yourself using ExchangeBindings (or even Federation). We could look at supporting exchange bindings and do have an issue around it, but have not seen the use cases yet.

https://www.rabbitmq.com/e2e.html

**Reply by remotenode** [reply]
@iancooper thank you for your answer. 
Let me give a bit more details:
Let's say I want to post two messages:
1) OrderCreated
2) UserCreated

I need these messages to be posted to different exchanges, I don't need an exchange to exchange bindings:)

And in my case these exchanges should be fanout exchanges, I want to use fanout exchanges just because they are faster than direct exchanges.

**Reply by iancooper** [reply]
Ah right. So you want to publish to different exchanges. In essence you would need two different command processors right now, we don't support an exchange per routing key model.

it is a reasonable point. I see a lot of different ways to decide how many exchanges you want from one per system to one per microservice (the model we mostly work with), and i guess that one per ' channel' might make sense with a fanout exchange based approach.

Feel free to drop an issue in with a feature request, post v9 getting out the door we could probably tackle it in the next quarter.

But right now, it's a command processor per exchange.;

Sorry about that
